geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZIncrByIntegrationTest.java

```diff
@@ -0,0 +1,290 @@
+    jedis.zadd(KEY, POSITIVE_INFINITY, MEMBER);
+
+    assertThatThrownBy(() -> jedis.zincrby(KEY, NEGATIVE_INFINITY, MEMBER))
+        .hasMessageContaining(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
+  }
+
+  @Ignore("blocked by GEODE-9317")
+  @Test
+  public void incrementingScoreOfNegativeInfinityByPositiveInfinity_throwsNaNError() {
+    jedis.zadd(KEY, NEGATIVE_INFINITY, MEMBER);
+
+    assertThatThrownBy(() -> jedis.zincrby(KEY, POSITIVE_INFINITY, MEMBER))
+        .hasMessageContaining(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
+  }
+
+  /************* key or member does not exist *************/
+  @Test
+  public void shouldCreateNewKey_whenIncrementedKeyDoesNotExist() {
+    final double increment = 1.5;
+
+    assertThat(jedis.zincrby(KEY, increment, MEMBER)).isEqualTo(increment);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Test
+  public void shouldCreateNewMember_whenIncrementedMemberDoesNotExist() {
+    final double increment = 1.5;
+    jedis.zadd(KEY, increment, "something".getBytes());
+
+    assertThat(jedis.zincrby(KEY, increment, MEMBER)).isEqualTo(increment);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  /************* happy path *************/
+  @Test
+  public void shouldIncrementScore_givenKeyAndMemberExist() {
+    testZIncrByIncrements(KEY, MEMBER, 2.7, 1.5);
+  }
+
+  @Test
+  public void shouldIncrementScore_givenNegativeIncrement() {
+    testZIncrByIncrements(KEY, MEMBER, 2.7, -1.5);
+  }
+
+  @Test
+  public void shouldIncrementScoreToANegativeNumber_givenNegativeIncrementGreaterThanScore() {
+    testZIncrByIncrements(KEY, MEMBER, 2.7, -5.6);
+  }
+
+  @Test
+  public void shouldIncrementScore_givenNegativeInitialScoreAndPositiveIncrement() {
+    testZIncrByIncrements(KEY, MEMBER, -2.7, 1.7);
+  }
+
+  @Test
+  public void shouldIncrementScore_givenIncrementInExponentialForm() {
+    testZIncrByIncrements(KEY, MEMBER, -2.7, 2e5);
+    testZIncrByIncrements(KEY, MEMBER, 2e5, -2.7);
+  }
+
+  /************* concurrency *************/
+  @Test
+  public void testConcurrentZIncrBy_performsAllZIncrBys() {
+    AtomicDouble expectedValue = new AtomicDouble(0.0);
+
+    jedis.zadd(STRING_KEY, expectedValue.get(), STRING_MEMBER);
+
+    new ConcurrentLoopingThreads(1000,
+        (i) -> {
+          double increment = ThreadLocalRandom.current().nextDouble(-50, 50);
+          expectedValue.addAndGet(increment);
+          jedis.zincrby(STRING_KEY, increment, STRING_MEMBER);
+        },
+        (i) -> {
+          double increment = ThreadLocalRandom.current().nextDouble(-50, 50);
+          expectedValue.addAndGet(increment);
+          jedis.zincrby(STRING_KEY, increment, STRING_MEMBER);
+        }).run();
+
+    Offset<Double> offset = Offset.offset(Math.abs(expectedValue.get()) * 0.05);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isCloseTo(expectedValue.get(), offset);
```

Comment:
Is there a reason that this can't just be 
```
assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(expectedValue.get());
```
Given the test name, if the score is only close to the expected value rather than exactly equal, we didn't perform all the `zincrby`s and so the test should fail.

Reply:
yeah I have no idea what that was about. good catch